src/zenml/integrations/kubeflow/orchestrators/kubeflow_component.py

```diff
@@ -153,38 +152,33 @@ def __init__(
             arguments.append(_encode_runtime_parameter(param))
 
         stack = Repository().active_stack
-        artifact_store = stack.artifact_store
-        metadata_store = stack.metadata_store
 
+        # go through all stack components and identify those that advertise
+        # a local path where they persist information that they need to be
+        # available when running pipelines. For those that do, mount them
+        # into the Kubeflow container.
         has_local_repos = False
-
-        if isinstance(artifact_store, LocalArtifactStore):
-            has_local_repos = True
-            host_path = k8s_client.V1HostPathVolumeSource(
-                path=artifact_store.path, type="Directory"
-            )
-            volumes[artifact_store.path] = k8s_client.V1Volume(
-                name="local-artifact-store", host_path=host_path
-            )
-            logger.debug(
-                "Adding host path volume for local artifact store (path: %s) "
-                "in kubeflow pipelines container.",
-                artifact_store.path,
-            )
-
-        if isinstance(metadata_store, SQLiteMetadataStore):
+        for stack_comp in stack.components.values():
+            local_path = stack_comp.local_path
```

Comment:
I saw this loop in at least two places, do you think it makes sense to sort of aggregate them in a 
```
@property
def local_paths(self) -> Set[str]
```
or similar on the `Stack`?

Reply:
This looked like a good idea initially, but as I started to implement it I realized that I also need the components associated with the paths, for logging related tasks.